.github/workflows/ci-ipfs.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy IPFS Base
+
+on:
+  workflow_call:
+    inputs:
+      ipfs_folder:
+        description: 'IPFS build folder'
+        default: 'out/'
+        type: string
+      W3S_TOKEN:
+        description: 'web3storage token'
+        type: string
+      GW3_TOKEN:
+        description: 'gateway3 token'
+        type: string
+
+permissions:
+  contents: write
+
+jobs:
+  deploy-ipfs:
+    runs-on: ubuntu-latest
+    name: Build and deploy
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+      - name: Extract branch name
+        id: extract_branch
```

Comment:
it should be an argument

Reply:
How? A workflow can be dispatched on any branch.

Reply:
you can restrict it in the calling workflow